### ISSUE TITLE
Have ``RemoteDispatcher`` support ``*`` in prefix description

### ISSUE THREAD

**Issue by CJ-Wright** [NSLS-II/bluesky#1166]
It might be helpful to be able to write ``d = RemoteDispatcher('localhost:1234', prefix='xpd_*')`` to listen to all things with the pre prefix `xpd`. This would allow us to have behavior like the current ``prefix=None`` (which consumes all documents headed its way) but with more specificity.

## Expected Behavior
``*`` is a valid thing in a prefix, and means all documents coming from any prefix which matches my string (with wildcards).

## Current Behavior
Wildcards in prefix is not supported

## Possible Solution
Regex the prefix?

## Context
This will hep with multiplexing message buses and will help when we eventually nail down a convention for prefix/topic names. 

CC @stuartcampbell since we were discussing topic names

**Comment by danielballan** [comment]
I agree that wildcards is enough. (Regex seems like overkill.) It's also how pattern matching works in Kibana/Elastic, which seems good.

You asked on Slack for help with the implementation. Does this snippet help?

```py
In [1]: prefix = b'xpd*'

In [2]: messages = [b"xpd1 start {...}", b"xpd2 start {...}", b"pdf start {...}"]

In [3]: for msg in messages:
   ...:     msg_prefix, *_ = msg.split(b' ', 1)
   ...:     if prefix.endswith(b'*'):
   ...:         if msg_prefix.startswith(prefix[:-1]):
   ...:             print(msg)
   ...:     elif msg_prefix == prefix:
   ...:         print(msg)
   ...:         
b'xpd1 start {...}'
b'xpd2 start {...}'
```

**Comment by CJ-Wright** [comment]
Should we match kafka? https://github.com/dpkp/kafka-python/blob/68c8fa4ad01f8fef38708f257cb1c261cfac01ab/kafka/consumer/group.py#L728

On the implementation side I was mostly interested in if we wanted to keep the prefixes as bytes or handle the encoding/decoding/matching later.
We could put wildcards in the middle too I suspect.
